src/test/java/org/jenkinsci/plugins/oic/OicSecurityRealmTokenExpirationTest.java

```diff
@@ -49,9 +60,14 @@ void handleTokenExpiration_noUser() throws Exception {
         final TestRealm realm = new TestRealm.Builder(wireMock).build();
         jenkins.setSecurityRealm(realm);
 
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
         try (MockedStatic<User> userMocked = mockStatic(User.class)) {
             userMocked.when(() -> User.get2(any())).thenReturn(null);
-            assertTrue(realm.handleTokenExpiration(
+
+            User user = User.get2(authentication);
+            Assertions.assertNull(user);
```

Comment:
was this for some debugging or some other reason - if we are using a JenkinsRule per test then the context should be reset correctly.
Generally because the Authentication should be per Thread and handled by the server framework.  if it is not reset then a test is not clearing up after doing something questionable. (or the production code is doing something questionable)

Reply:
(generally avoid mixing WithJenkins/JenkinsRule and Mockito.)

Reply:
> (generally avoid mixing WithJenkins/JenkinsRule and Mockito.)

How to avoid this mixing if I need to mock `jakarta.servlet.http.HttpServletRequest` resp.
`jakarta.servlet.http.HttpServletResponse`

Reply:
use a real web request and implement some test API to invalidate (expire) tokens.

or don't use JenkinsRule and use a mocking to mock enough or what you need to test the method.